pkg/downloader/custom_github_detector.go

```diff
@@ -0,0 +1,110 @@
+package downloader
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	log "github.com/charmbracelet/log"
+	"github.com/cloudposse/atmos/pkg/schema"
+)
+
+// Predefined errors for better reuse and static analysis.
+var (
+	ErrInvalidGitHubURL = errors.New("invalid GitHub URL")
+	ErrURLEmpty         = errors.New("source URL is empty")
+	ErrFailedParseURL   = errors.New("failed to parse URL")
+)
+
+type EnvProvider func(key string) string
+
+// Logger interface allows logging abstraction for testing.
+type Logger interface {
+	Debug(any, ...any)
+}
+
+// customGitHubDetector detects and modifies GitHub URLs.
+type customGitHubDetector struct {
+	AtmosConfig *schema.AtmosConfiguration
+	GetEnv      EnvProvider
+	Log         Logger
+}
+
+// NewCustomGitHubDetector initializes a new detector with default dependencies.
+func NewCustomGitHubDetector(config *schema.AtmosConfiguration) *customGitHubDetector {
+	return &customGitHubDetector{
+		AtmosConfig: config,
+		GetEnv:      os.Getenv,
+		Log:         log.Default(),
+	}
+}
+
+// Detect checks if a URL is a GitHub repo and injects authentication if needed.
+func (d *customGitHubDetector) Detect(src, _ string) (string, bool, error) {
+	if src == "" {
+		return "", false, ErrURLEmpty
+	}
+
+	if !strings.Contains(src, "://") {
+		src = "https://" + src
+	}
+
+	parsedURL, err := url.Parse(src)
+	if err != nil {
+		d.Log.Debug(fmt.Sprintf("Failed to parse URL %q: %v", src, err))
+		return "", false, fmt.Errorf("failed to parse URL %q: %w", src, err)
+	}
+
+	if strings.ToLower(parsedURL.Host) != "github.com" {
+		d.Log.Debug(fmt.Sprintf("Host is %q, not 'github.com', skipping token injection", parsedURL.Host))
```

Comment:
Heh, you hacked around the lint check to ensure no format specifiers are used. The check was supposed to ensure you use semantic logging.

Reply:
Ohh okay now i got it. Thanks

Reply:
Done